libcontainer/nsenter/nsenter_go121.go

```diff
@@ -0,0 +1,78 @@
+//go:build linux && !gccgo && go1.21
+// +build linux,!gccgo,go1.21
+
+package nsenter
+
+// Since Go 1.21 <https://github.com/golang/go/commit/c426c87012b5e>, the Go
+// runtime will try to call pthread_getattr_np(pthread_self()). This causes
+// issues with nsexec and requires backward compatibility with old versions
+// of the glibc.
+
```

Comment:
Perhaps you should mention that the code below is a modified version of code in src/runtime/cgo/gcc_stack_unix.c in golang sources, and briefly describe what was modified.

Reply:
Might also make sense to remove solaris/illumos support, for clarity.